includes/data-stores/class-wc-product-variation-data-store-cpt.php

```diff
@@ -165,6 +164,23 @@ public function update( &$product ) {
 	*/
 
 	/**
+	 * Generates a title with attribute information for a variation.
+	 * Products with one attribute will get a title of the form "Name - Value"
+	 * Products with 2+ attributes will get a title of the form "Name - Attribute: Value, Attribute: Value"
+	 *
+	 * @since 2.7.0
+	 * @param WC_Product
+	 * @return string
+	 */
+	protected function generate_product_title( $product ) {
+		$use_longform_title = apply_filters( 'woocommerce_product_variation_title_use_longform_style', count( $product->get_attributes() ) > 1, $product );
```

Comment:
I'm not sure 'use_longform_style' is clear enough. How about describing what we're actually doing? i.e. `woocommerce_product_variation_title_include_attribute_names`. True if we include them here.